### DISCUSSION TITLE
brew services display unintuitive

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#2807] [Getting started]
Asking because I just used `brew services` for the first time and seeing unexpected behavior. I am using it with `dnsmasq` and `stubby`.

I start both services:
`brew services start stubby`
`sudo brew services start dmsmasq`

Then run `brew services` again I would expect to see the status of both services, but can only see one no matter if I am `user`, `admin`, `root`. User has admin privileges and is the same user that installed homebrew.

**Is there something amiss or is this expected behavior?**

```
01/17/22_20:18:21 /Users/john
0-zsh% brew services
Name            Status  User File
dbus            none         
dnsmasq         none    root 
emacs           none         
selenium-server none         
stubby          started john ~/Library/LaunchAgents/homebrew.mxcl.stubby.plist
unbound         none         

01/17/22_20:18:28 /Users/john
0-zsh% sudo brew services
Name            Status  User File
dbus            none         
dnsmasq         started root /Library/LaunchDaemons/homebrew.mxcl.dnsmasq.plist
emacs           none         
selenium-server none         
stubby          none    root 
unbound         none         

01/17/22_20:18:44 /Users/john
0-zsh% login root
Password:
iMac2019_eth:~ root# brew services 
Name            Status  User File
dbus            none         
dnsmasq         started root /Library/LaunchDaemons/homebrew.mxcl.dnsmasq.plist
emacs           none         
selenium-server none         
stubby          none         
unbound         none         
iMac2019_eth:~ root# exit
logout
```

**Comment by SMillerDev** [comment]
`sudo` means "run this as root" and users can't read what root is running for security reasons so it also can't display what is running as root.

**Reply by ghost** [reply]
I can see what plists are running with Lingon X for all users on my system. Modifying it is a different story - I do need to enter a password.

If you take a look at this [page](https://chabik.com/2020/05/dot-with-stubby-and-dnsmasq-on-macos/) (which I used as a guide to install `dnsmasq` and `stubby`) he is able to see the status of both `dnsmasq` and `stubby` in brew services... at the same time.

```
brew services
Name           Status  User   Plist
dnsmasq        stopped
stubby         stopped

brew services start stubby
==> Successfully started `stubby` (label: homebrew.mxcl.stubby)

sudo brew services start dnsmasq
Warning: Taking root:admin ownership of some dnsmasq paths:
  /usr/local/Cellar/dnsmasq/2.81/sbin
  /usr/local/Cellar/dnsmasq/2.81/sbin/dnsmasq
  /usr/local/opt/dnsmasq
  /usr/local/opt/dnsmasq/sbin
  /usr/local/var/homebrew/linked/dnsmasq
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
==> Successfully started `dnsmasq` (label: homebrew.mxcl.dnsmasq)

brew services
Name           Status  User   Plist
dnsmasq        started root   /Library/LaunchDaemons/homebrew.mxcl.dnsmasq.plist
stubby         started hadret /Users/hadret/Library/LaunchAgents/homebrew.mxcl.stubby.plist
```

Maybe he has changed his sudoers file and added `dnsmasq` to his user `hadret` ?

**Reply by SMillerDev** [reply]
Or it was written on an earlier macOS version with different security. Or its just an illustration. There's not much I can determine based on a guide.

**Reply by carlocab** [reply]
It might be worth tweeting hadret to ask them.
